identity.go

```diff
@@ -0,0 +1,223 @@
+	generatedAppID := newAppId(cfg.AppSecret)
+
+	if !bytes.Equal(generatedAppID, cfg.AppID) {
+		return identity{}, errors.New("app secret and app ID mismatch")
+	}
+
+	userID := hashUserID(cfg.AppID, userIDString)
+	userSecret := newUserSecret(userID)
+
+	epubSignKey, eprivSignKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return identity{}, err
+	}
+
+	payload := make([]byte, len(epubSignKey) + len(userID))
+	copy(payload, epubSignKey)
+	copy(payload[len(epubSignKey):], userID)
+
+	delegationSignature := ed25519.Sign(cfg.AppSecret, payload)
+
+	id := identity{
+		publicIdentity: publicIdentity{
+			TrustchainID: cfg.AppID,
+			Target:       TargetUser,
+			Value:        base64.StdEncoding.EncodeToString(userID),
+		},
+		DelegationSignature:          delegationSignature,
+		EphemeralPrivateSignatureKey: eprivSignKey,
+		EphemeralPublicSignatureKey:  epubSignKey,
+		UserSecret:                   userSecret,
+	}
+
+	return id, nil
+}
+
+func newProvisionalIdentity(cfg config, target Target, value string) (provisionalIdentity, error) {
+	publicSignatureKey, privateSignatureKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return provisionalIdentity{}, err
+	}
+	publicEncryptionKey, privateEncryptionKey, err := NewKeyPair()
+	if err != nil {
+		return provisionalIdentity{}, err
+	}
+
+	provId := provisionalIdentity{
+		publicProvisionalIdentity: publicProvisionalIdentity{
+			publicIdentity: publicIdentity{
+				TrustchainID: cfg.AppID,
+				Target:       target,
+				Value:        value,
+			},
+			PublicEncryptionKey: publicEncryptionKey,
+			PublicSignatureKey:  publicSignatureKey,
+		},
+		PrivateSignatureKey:  privateSignatureKey,
+		PrivateEncryptionKey: privateEncryptionKey,
+	}
+
+	return provId, nil
+}
+
+func hashProvisionalIdentityEmail(email string) (hash string) {
+	hashedValue := blake2b.Sum256([]byte(email))
+	return base64.StdEncoding.EncodeToString(hashedValue[:])
+}
+
+func hashProvisionalIdentityValue(value string, privateSignatureKeyB64 string) (hash string) {
+	privateSignatureKey, _ := base64.StdEncoding.DecodeString(privateSignatureKeyB64)
+	hashSalt := blake2b.Sum256(privateSignatureKey)
+	hashedValue := blake2b.Sum256(append(hashSalt[:], []byte(value)...))
+	return base64.StdEncoding.EncodeToString(hashedValue[:])
+}
+
+// Create creates a new identity usable for interaction with the Tanker server
+func Create(config Config, userID string) (string, error) {
+	return New(config, userID)
+}
+
+// CreateProvisional creates a new provisional identity usable for interaction with the Tanker server
+func CreateProvisional(config Config, target Target, value string) (string, error) {
+	return NewProvisional(config, target, value)
+}
+// GetPublicIdentity returns the public identity derived from the given identity
+func GetPublicIdentity(b64Identity string) (string, error) {
+	return GetPublic(b64Identity)
+}
+
+func New(cfg Config, userID string) (string, error) {
```

Comment:
Do we need `New` and `Create` ?

Reply:
I added `New` because it seems to me that it is more of an idiomatic way to Go, and left `Create` for backwards compatibility. `Create` is just an alias (at the beginning I even wanted to deprecate `Create` but that finally seemed like a bad idea ... :sweat_smile: )